proposals/NNNN-widget-support.md

```diff
@@ -0,0 +1,248 @@
+
+Still widgets have affect to the HMI level of the app's main screen. Depending on the app policies:
+
+1. Apps can create widgets from any HMI level as allowed by policies (e.g. from HMI_NONE or BACKGROUND) and send `Show` or `SetDisplayLayout` to add content (text and soft buttons) to that widget. The RPCs sent to the widget follow policies of HMI_LIMITED
+2. If a user taps on a soft button in the widget and the app main screen is in HMI_NONE the app will automatically transit to HMI_BACKGROUND
+3. Exception 1: If the app is a navigation or media app the app will transit to HMI_LIMITED instead. Navigation apps will then be ready to start navigation by starting video and audio serivce. This exception will also make media apps audible.
+4. Exception 2: If the app specified the soft button with `STEAL_FOCUS` system action the app will transit to HMI_FULL and the main screen will be shown
+5. If a user taps anywhere in the widget but not on a soft button the app will transit to HMI_FULL and the main screen will be shown
+
+#### PredefinedScreens
+
+The enum `PredefinedScreens` specifies what screens and IDs are predefined and precreated on behalf of the app.
+
+The default screen is always available and represents the app screen on the main display. It's an equivalent to the todays app screen. For backward compatibility this will ensure the app always has at least the default screen on the main display. The app can choose to use this predefined enum element to specifically address app's main screen or to duplicate screen content. It is not possible to duplicate another screen to the default screen.
+
+### Screen manipulation
+
+Working on this proposal we grouped all RPCs into three categories: system, display or screen related. The RPC `Show` and `SetDisplayLayout` are identified as the only screen related RPCs. In order to allow an app to address a specific screen these two RPCs need to have an option parameter for the screen ID.
+
+```xml
+<function name="Show" functionID="ShowID" messagetype="request" since="1.0">
+ :
+ :
+<param name="screenID" type="Integer" mandatory="false" since="5.1" />
+</function>
+
+<function name="SetDisplayLayout" functionID="SetDisplayLayoutID" messagetype="request" since="2.0">
+ :
+ :
+ <param name="screenID" type="Integer" mandatory="false" since="5.1" />
+</function>
+```
+
+This allows the app to use the full flexibility of the Show RPC in other screens.
+
+### Screen templates
+
+Today SDL comes with a set of templates available the app can choose from. The information what templates are available to the app is provided in 
```

Comment:
Update to:
**Today, SDL comes with a set of templates available for the app developer to choose from. The information for what templates are available for the apps are provided in**